src/renderer/certificate/lens-proxy-certificate.injectable.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type { SelfSignedCert } from "selfsigned";
+import { lensProxyCertificateChannel } from "../../common/certificate/lens-proxy-certificate-channel";
+import { lensProxyCertificateInjectionToken } from "../../common/certificate/lens-proxy-certificate-injection-token";
+import requestFromChannelInjectable from "../utils/channel/request-from-channel.injectable";
+
+const lensProxyCertificateInjectable = getInjectable({
+  id: "lens-proxy-certificate",
+  instantiate: (di) => {
+    const requestFromChannel = di.inject(requestFromChannelInjectable);
+
+    let certState: SelfSignedCert;
+    const cert = {
+      get: () => {
+        if (!certState) {
+          throw "certificate has not been set";
+        }
+
+        return certState;
+      },
+      set: (certificate: SelfSignedCert) => {
+        certState = certificate;
+      },
+    };
+    
+    requestFromChannel(lensProxyCertificateChannel).then((value) => {
+      cert.set(value);
+    });
```

Comment:
Shouldn't this to be `await`ed before returning `cert`?

Reply:
We cannot await because `lensProxyCertificateInjectionToken` interface does not return a promise. If it would be async then we would need to refactor a big chunk of logic in `main` process because it currently does not allow to await. IMHO we can do that in a separate PR if there is a bigger need.

Reply:
> Actually I think this is the wrong place for this entirely. This should be a runnable. Probably under the `beforeFrameStartsFirstInjectionToken` phase

I agree, will refactor.

Reply:
Fixed.